core/domain/acl_decorators.py

```diff
@@ -72,40 +71,286 @@ def check_activity_accessible(
             activity_rights.viewable_if_private)
 
 
-def check_exploration_editable(user_id, user_actions, exploration_id):
-    """Returns a boolean to signify whether given exploration is editable
-    by the user or not.
+def check_can_edit_activity(
+        user_id, user_actions, activity_type, activity_id):
+    """Checks whether the user can edit given activity.
 
     Args:
         user_id: str. Id of the given user.
```

Comment:
oh. On a second note. I feel I am uselessly checking user_id for the None case.

Since user_id is used only for is_owner etc which will return correct answer even in case of None so it need not be checked explicitly.

So, should I remove it? (Also, if I remove the case of None, do I have to modify documentation?)